common/lib/helpers.py

```diff
@@ -887,6 +905,51 @@ def _sets_to_lists_gen(d):
 
     return dict(_sets_to_lists_gen(d))
 
+
+def url_to_hash(url, remove_scheme=True, remove_www=True):
+    """
+    Convert a URL to a filename; some URLs are too long to be used as filenames, this keeps the domain and hashes the
+    rest of the URL.
+    """
+    parsed_url = urlparse(url.lower())
+    if parsed_url:
+        if remove_scheme:
+            parsed_url = parsed_url._replace(scheme="")
+        if remove_www:
+            netloc = re.sub(r"^www\.", "", parsed_url.netloc)
+            parsed_url = parsed_url._replace(netloc=netloc)
+
+        url = re.sub(r"[^0-9a-z]+", "_", urlunparse(parsed_url).strip("/"))
+    else:
+        # Unable to parse URL; use regex
+        if remove_scheme:
+            url = re.sub(r"^https?://", "", url)
+        if remove_www:
+            if not remove_scheme:
+                scheme = re.match(r"^https?://", url).group()
+                temp_url = re.sub(r"^https?://", "", url)
+                url = scheme + re.sub(r"^www\.", "", temp_url)
+            else:
+                url = re.sub(r"^www\.", "", url)
+
+        url = re.sub(r"[^0-9a-z]+", "_", url.lower().strip("/"))
+
+    return hashlib.blake2b(url.encode("utf-8"), digest_size=24).hexdigest()
+
+def validate_url(x):
```

Comment:
Likewise, `ural` has `is_url()` (and some variations): https://github.com/medialab/ural?tab=readme-ov-file#is_url

Reply:
These functions were used by extensions and just deleting them breaks those. Fine with replacing them with whatever you like.

Perhaps helper functions should not be used by extensions, but the extensions are going to rely on 4CAT functions elsewhere so we need some sort of guiding principle as to what an extension can expect 4CAT to keep.